src/machine-learning/machine-learning.ts

```diff
@@ -171,23 +187,23 @@ export class MachineLearning implements FirebaseServiceInterface {
     return this.client.deleteModel(modelId);
   }
 
-  private convertOptionstoContent(options: ModelOptions, forUpload?: boolean): Promise<ModelContent> {
-    const modelContent = deepCopy(options);
+  private maybeSignUrl(options: ModelOptions, forUpload?: boolean): Promise<ModelOptions> {
```

Comment:
`maybe` sounds like it's non-deterministic. Perhaps just `signUrlIfPresent()`?

Also is `forUpload` ever `false`?

Reply:
done